datalad/core/distributed/clone.py

```diff
@@ -351,6 +371,74 @@ def __call__(
                 ds.subdatasets(path, set_property=[("datalad-url", source)])
 
 
+def _get_url_mappings(cfg):
+    cfg_prefix = 'datalad.clone.url-substitute.'
+    # figure out which keys we should be looking for
+    # in the active config
+    subst_keys = set(k for k in cfg.keys() if k.startswith(cfg_prefix))
+    # and in the common config specs
+    from datalad.interface.common_cfg import definitions
+    subst_keys.update(k for k in definitions if k.startswith(cfg_prefix))
+    # TODO a potential sorting of substitution series could be implemented
+    # here
```

Comment:
I guess we would need to introduce `datalad.clone.url-substitute.<series>.priority` or alike?

Reply:
That, or sorting based on series ID, or inspection of the regex match specification (see "longest match" comment). I think there are multiple possible ways forward.